yetanotherco/aligned_layer
# feat: verify contracts on upgrade

To test this, you can:
1) `make deploy_aligned_contracts`
2) check it is verified on [holesky etherscan](https://holesky.etherscan.io/)
3) `make upgrade_aligned_contracts`
4) check it is still verified on [holesky etherscan](https://holesky.etherscan.io/)

Also, to simulate what will happen in prod, you can: 
1) `git checkout main`
2) `make deploy_aligned_contracts`
3) check it is **not** verified on [holesky etherscan](https://holesky.etherscan.io/)
4) `git checkout 495-feat-verify-contracts-on-etherscan`
5) `make upgrade_aligned_contracts`
6) check it is now verified on [holesky etherscan](https://holesky.etherscan.io/)

Note: The proxy will appear as without endpoints, you must say to etherscan it is a proxy, as follows:
1) click on "is this a proxy?)
<img width="1222" alt="image" src="https://github.com/yetanotherco/aligned_layer/assets/43704209/013892f6-ce1a-449a-a687-1f02cf368871">

2) verify
<img width="701" alt="image" src="https://github.com/yetanotherco/aligned_layer/assets/43704209/d7bce8f6-1f8e-45bc-abfe-4f15961fc288">

3) save
<img width="522" alt="image" src="https://github.com/yetanotherco/aligned_layer/assets/43704209/6f0d6052-ce27-4ee9-bfb7-55df205aafac">

4) feel free to return to your proxy
<img width="644" alt="image" src="https://github.com/yetanotherco/aligned_layer/assets/43704209/ab4b8b09-da86-4a1c-8da1-32fa4783a746">

5) under contracts, new tabs "read as proxy/write as proxy" should appear
<img width="619" alt="image" src="https://github.com/yetanotherco/aligned_layer/assets/43704209/da52f6b5-eb6f-4375-9f15-7fe75a5f3b1e">

6) now you should be able to interact with the contract through the proxy
<img width="424" alt="image" src="https://github.com/yetanotherco/aligned_layer/assets/43704209/e2b9cb5e-b830-4315-8208-37f6558bfe7c">
